src/ast.rs

```diff
@@ -6,6 +6,34 @@ use std::net::IpAddr;
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug)]
+pub struct Route {
```

Comment:
Do you think we should use a more specific name? `Router`, `Route`, `RouteTerm`, and `RouteLogicalOperators`, which looks weird.

Reply:
The name of 'Route' is derived from the existing 'Router' container. The meaning is that Router container is a collection of individual Route.  And the RouteTerm is a subcomponent of a Route item.

Reply:
fixed